packages/webapp/src/containers/Filter/types.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+// In tempStateReducer > filterReducer
+export interface ReduxFilterEntity {
+  [filterKey: /* e.g. STATUS, LOCATION */ string]: FilterState;
+}
+
+export interface FilterState {
+  [filterOptionId: string]: /* e.g. ABANDONED, COMPLETED */ {
+    active: boolean;
+    label: string /* e.g. 'Abandoned', 'Completed */;
+  };
+}
+
+// (filterKey, filterState) => setTypesFilter({ ...typesFilter, [filterKey]: filterState })
+export type FinanceReportOnChangeCallback = (filterKey: string, filterState: FilterState) => void;
+
+// setDirty: () => !isDirty && setIsDirty(true)
+export type FilterPageOnChangeCallback = () => void;
```

Comment:
No, I'll do as @antsgar suggested above, but I'm still looking for the 100% pleasing solution to convey that sometimes this is just `setDirty()`!  Any thoughts? Does the chain of onChange read okay to you overall and in the component file (as a previous consumer of the filter pattern when building the sales hook)?

I know I didn't actually type the component you used (since we're not touching it anymore this sprint).